mini_pupper_maze/node/ai_image_service.py

```diff
@@ -0,0 +1,152 @@
+        """
+        logging.debug("ai_image_response start!")
+        ms_start = int(time.time() * 1000)
+
+        buffered = BytesIO()
+        image.save(buffered, format="JPEG")
+        image_bytes = buffered.getvalue()
+
+        image_base64 = base64.b64encode(image_bytes).decode('utf-8')
+        image_data_url = f"data:image/jpeg;base64,{image_base64}"
+
+        image_message = {
+            "type": "image_url",
+            "image_url": {
+                "url": image_data_url
+            }
+        }
+        text_message = {"type": "text", "text": text}
+
+        message = HumanMessage(content=[text_message, image_message])
+
+        output = llm.invoke([message])
+
+        #logging.debug(f"ai_image_response response: {output}")
+        result = output.content
+        logging.debug(f"text response: {result}")
+        ms_end = int(time.time() * 1000)
+        logging.debug(f"ai_image_response end, delay = {ms_end - ms_start}ms")
+        return result
+
+    def image_recognition():
+
+        direction_input_prompt="""
+        You are an expert in identifying the degree of black line from the image
+                    and tell how to move along the line
+                    in below format
+
+                    [
+                        'move_forward',
+                        'look_middle',
+                        'look_down',
+                        'look_up',
+                        'move_backward',
+                        'look_left',
+                        'look_right',
+                        'move_left',
+                        'move_right',
+                        'move_forward',
+                        'move_backward',
+                        'look_middle',
+                        'look_left',
+                        'look_right'
+                    ]
+
+        """
+
+        sharp_direction_input_prompt="""
+        You are an expert in identifying pointers of different direction from the image
+                    and tell what directions are they
+                    in below format only if you see a pointer pointing to the right and output a "." if you dont see a pointer
+
+                    [
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                        'move_right',
+                    ]
+        """
+
+        multi_model = ChatVertexAI(model="gemini-pro-vision")
+
+        image = take_photo()
+        image = image.resize((320, 240))
+        direction_response = ai_image_response(multi_model, image=image, text=direction_input_prompt)
```

Comment:
I think this takes some computation power for a function call to ai_image_response (line137-138). Suggest moving it into the if-else block